omnibus/config/software/openssl-fips.rb

```diff
@@ -0,0 +1,54 @@
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com).
+# Copyright 2016-present Datadog, Inc.
+
+# Embedded OpenSSL to meet FIPS requirements. It comes in two parts:
+# 1. The FIPS module itself . It must use a FIPS-validated version 
+#    and follow the build steps outlined in the OpenSSL FIPS Security Policy.
+# 2. The OpenSSL library (this software definition), which can be any 3.0.x version. This library will use the FIPS provider. 
+
+name "openssl-fips"
+default_version "0.0.1"
+
+OPENSSL_VERSION="3.0.15"
+OPENSSL_SHA256_SUM="23c666d0edf20f14249b3d8f0368acaee9ab585b09e1de82107c66e1f3ec9533"
+OPENSSL_FILENAME="openssl-#{OPENSSL_VERSION}"
+
+DIST_DIR="#{install_dir}/embedded"
+
+source url: "https://github.com/openssl/openssl/releases/download/#{OPENSSL_FILENAME}/#{OPENSSL_FILENAME}.tar.gz",
+           sha256: "#{OPENSSL_SHA256_SUM}",
+           extract: :seven_zip,
+           target_filename: "#{OPENSSL_FILENAME}.tar.gz"
+
+relative_path "openssl-#{OPENSSL_VERSION}"
+
+dependency "openssl-fips-provider"
+
+build do
+    command "./Configure --prefix=\"#{DIST_DIR}\" \
+                --libdir=lib \
+                -Wl,-rpath=\"#{DIST_DIR}/lib\" \
+                no-asm no-ssl2 no-ssl3 \
```

Comment:
If modifying `omnibus-software`'s `openssl3.rb` is not an option, is `no-asm` really needed here? I fear that will hurt the performances significantly

Reply:
I don't have the context on why we used no-asm in the fips-proxy but I guess this can be removed